extensions/objects/models/objects.py

```diff
@@ -59,7 +59,81 @@ def normalize(cls, raw):
         Raises:
             TypeError. The Python object cannot be normalized.
         """
-        return schema_utils.normalize_against_schema(raw, cls.SCHEMA)
+        return schema_utils.normalize_against_schema(raw, cls.get_schema())
+
+
+class BaseTranslatableObject(BaseObject):
+    """Base translatable object class.
+
+    This is a superclass for objects that are translatable and thus require a
+    content id. This class enforces that the object is a dictionary with a
+    content id field. The schema of actual value is determined by the
+    _get_value_schema() method.
+    """
+
+    @staticmethod
+    def _get_value_schema():
+        """Returns a list of properties that store the object value.
+
+        Returns:
+            list(dict). A list of properties that store the object value.
+        """
+        raise NotImplementedError('Please Implement this method')
+
+    @staticmethod
+    def _normalize_value(raw):
+        """Returns a method that normalizes the values of the object.
+
+        Args:
+            raw: *. A translatable Python object whose values are to be
+                ormalized.
+
+        Returns:
+            dict. A normalized translatable Python object with its values
+            normalized.
+        """
+        raise NotImplementedError('Please Implement this method')
+
+    @classmethod
+    def get_schema(cls):
+        """Returns the object schema.
+
+        Returns:
+            dict. The object schema.
+        """
+        return {
+            'type': 'dict',
+            'properties': [{
+                'name': 'content_id',
+                # The default content id is none. However, it should be
+                # populated before being saved. The normalization method has
```

Comment:
The normalization method --> The normalize() method.

There is also an extra closing paren at the end of this sentence.